### PR TITLE
Adds deter mode support

### PR DESCRIPTION
**Pull request by ethan021021**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "vivintpy"
-version = "2023.3.3"
+version = "2023.3.4"
```
Comment (natekspencer) [inline pyproject.toml:3]: Let's not change the version in PR requests unless it is for the release of said version (there are two other places this has to change anyway to pass tests).

**Review by natekspencer — CHANGES_REQUESTED** [review]

**Comment by natekspencer** [comment]
Thank's for this, it looks good otherwise! I need to refactor the beam stuff, but I've been too lazy to do so.

**Comment by ethan021021** [comment]
@natekspencer I found another way we can check for `deter` on camera models. My python skills aren't the best but check out what I added :)

File: vivintpy/const.py [hunk 2]
```diff
@@ -99,6 +99,7 @@ class VivintDeviceAttribute:
     BYPASSED = "b"
     CAPABILITY = "ca"
     CAPABILITY_CATEGORY = "caca"
+    FEATURES = "fea"
```
Comment (natekspencer) [inline vivintpy/const.py:102]: Please put this in alphabetical order

File: vivintpy/devices/__init__.py [hunk 3]
```diff
@@ -60,6 +60,14 @@ def __init__(self, data: dict, alarm_panel: AlarmPanel | None = None) -> None:
...
             else None
         )
+        self._features = (
+            {
+                FeatureType(feature): feats.get(feature)
+                for feature in feats
+            }
+            if (feats := data.get(Attribute.FEATURES)) is not None
+            else None
+        )
```
Comment (natekspencer) [inline vivintpy/devices/__init__.py:70]: The features object appears to only include features if they are available (i.e. they are always "\<feature\>": true). Because there doesn't seem to be a full-fledged list which indicates features that are vs aren't supported, let's make this a list of supported features.

```json
"fea": {
  "video_thumbnails": true,
  "rb": true,
  "rstdef": true,
  "nght": true
}
```

File: vivintpy/devices/__init__.py [hunk 4]
```diff
@@ -102,6 +110,13 @@ def capabilities(
     ) -> dict[CapabilityCategoryType, list[CapabilityType]] | None:
         """Device capabilities."""
         return self._capabilities
+    
+    @property
+    def features(
+        self,
+    ) -> dict[str, bool] | None:
+        """Device Features."""
+        return self._features
```
Comment (natekspencer) [inline vivintpy/devices/__init__.py:119]: Please put this in alphabetical order

File: vivintpy/devices/camera.py [hunk 5]
```diff
@@ -106,6 +106,11 @@ def mac_address(self) -> str:
     def is_in_privacy_mode(self) -> bool:
         """Return True if privacy mode is active."""
         return bool(self.data[Attribute.CAMERA_PRIVACY])
+    
+    @property
+    def is_in_deter_mode(self) -> bool:
+        """Return True if deter mode is active."""
+        return bool(self.data[Attribute.DETER_ON_DUTY])
```
Comment (natekspencer) [inline vivintpy/devices/camera.py:113]: Please put this in alphabetical order

File: vivintpy/enums.py [hunk 6]
```diff
@@ -56,7 +56,23 @@ class CapabilityCategoryType(IntEnum):
     def _missing_(cls, value: object) -> CapabilityCategoryType:
         _LOGGER.debug("Unknown capability category type value: %s", value)
         return cls.UNKNOWN
+    
 
+class FeatureType(Enum):
```
Comment (natekspencer) [inline vivintpy/enums.py:61]: Please put this in alphabetical order in the file. Additionally, put the members in order alphabetically

File: vivintpy/enums.py [hunk 7]
```diff
@@ -56,7 +56,23 @@ class CapabilityCategoryType(IntEnum):
...
+    DYNAMIC_CHIMES = "dynamic_chimes_available"
+    VIDEO_THUMBNAILS = "video_thumbnails"
+    RB = "rb"
+    RST_DEF = "rstdef"
+    NGHT = "nght"
```
Comment (natekspencer) [inline vivintpy/enums.py:66]: RB should be REBOOT
RST_DEF should be RESTORE_DEFAULTS
NGHT should be SELECTABLE_NIGHTVISION

A few others I'm aware of:
HVAC_AWAY_SETBACK = "hvac_away_setback"
HVAC_SCHEDULES = "hvac_schedules"
LOCK_SELECTION = "lock_selection"
SMART_SENTRY_SNOOZE = "smart_sentry_snooze"
VEHICLE_DETECTION = "vehicle_detection"

**Review by natekspencer — CHANGES_REQUESTED** [review]
Also please format changed files with black/isort. If you install the dev dependencies, you should be able to run `tox -e mypy` and `tox -e lint` and see if you need to make changes before submitting.

File: vivintpy/devices/__init__.py (hunk 3 above)
Comment (ethan021021) [reply vivintpy/devices/__init__.py:70]: Oh huh I didn't realize that I'd imagine their API would include all features supported and can be toggled on/off by them remotely. If we were to assume these values are always `True` that could possibly cause an issue in the future?

File: vivintpy/enums.py (hunk 7 above)
Comment (ethan021021) [reply vivintpy/enums.py:66]: Will change. Had no idea what `RB` or `RST_DEF` was hah

File: vivintpy/devices/__init__.py (hunk 3 above)
Comment (natekspencer) [reply vivintpy/devices/__init__.py:70]: Yes, you should still only add the ones to the list that are "true" rather than assume. But we don't have an exhaustive list of devices/features and I've only seen values come back as true/false. So it doesn't really make sense to store a feature that is disabled in that list.

File: vivintpy/enums.py (hunk 7 above)
Comment (natekspencer) [reply vivintpy/enums.py:66]: I had pulled out a bunch of constants from the APK last year that I saved in a google sheet, and those just happened to be in there.

File: vivintpy/devices/__init__.py (hunk 3 above)
Comment (ethan021021) [reply vivintpy/devices/__init__.py]: Gotcha! Sometimes it's hard to parse what someone meant over text 👍

File: vivintpy/devices/__init__.py [hunk 8]
```diff
@@ -60,6 +66,14 @@ def __init__(self, data: dict, alarm_panel: AlarmPanel | None = None) -> None:
...
             else None
         )
+        self._features = (
+            {
+                FeatureType(feature): True
+                for feature in feats
+            }
+            if (feats := data.get(Attribute.FEATURES)) is not None
+            else None
+        )
```
Comment (natekspencer) [inline vivintpy/devices/__init__.py:76]: Again, let's make this a list of the features that are True, not a dict

File: vivintpy/devices/__init__.py [hunk 9]
```diff
@@ -108,6 +122,13 @@ def device_type(self) -> DeviceType:
         """Return the device type."""
         return DeviceType(self.data.get(Attribute.TYPE))
 
+    @property
+    def features(
+        self,
+    ) -> dict[FeatureType, bool] | None:
+        """Device Features."""
+        return self._features
+
```
Comment (natekspencer) [inline vivintpy/devices/__init__.py:131]: Fix the above, and then use `list[FeatureType]` instead of the dict here

**Review by natekspencer — CHANGES_REQUESTED** [review]

File: vivintpy/devices/__init__.py (hunk 8 above)
Comment (ethan021021) [reply vivintpy/devices/__init__.py]: Accidentally committed that.

File: vivintpy/devices/__init__.py [hunk 10]
```diff
@@ -70,6 +70,7 @@ def __init__(self, data: dict, alarm_panel: AlarmPanel | None = None) -> None:
             {
                 FeatureType(feature): True
```
Comment (ethan021021) [inline vivintpy/devices/__init__.py:71]: My python is quite rusty as you can tell, but this is what I was thinking of doing

Comment (ethan021021) [reply vivintpy/devices/__init__.py:71]: Nope. Nevermind

Comment (natekspencer) [reply vivintpy/devices/__init__.py:71]: Should be something like this:
```python
self._features = [FeatureType(feature) for feature in feats if feats.get(feature) is True] if (feats := data.get(Attribute.FEATURES)) is not None else None
```

Comment (ethan021021) [reply vivintpy/devices/__init__.py]: Give it a look now. Think I got it

**Comment by natekspencer** [comment]
@ethan021021 looks like you just have a couple of formatting issues now that need addressed:
https://github.com/natekspencer/vivintpy/actions/runs/5383216664/jobs/9769695918

**Comment by ethan021021** [comment]
Apologies as all of these tools are pretty new to me! I believe the latest commit should lint properly on the CI.

**Review by natekspencer — APPROVED** [review]
